#VULCAN-274/ Group reports in a page and minimise reports functionality

m-o-n-i-s-h opened the pull request:
# Group Reports and Minimize Reports Functionality
## Features
- Group one or more reports and we can specify a border and border colour to those grouped reports.
- Minimize button for reports.
- Added an option in table report config called compact canvas. when the query status is NO_DATA the card height is set to default 210px.
![MicrosoftTeams-image](https://github.com/neo4j-labs/neodash/assets/139316519/2a72564b-206d-4e07-9ada-624a61ca2ae3)
![MicrosoftTeams-image (2)](https://github.com/neo4j-labs/neodash/assets/139316519/8d2510ed-c407-456a-ab15-66f9187f7947)

Note: This implementation can be only be done through JSON for now. If you are okay with this implementation. Then I can create some component through which user can configure the group reports.
 
**Sample JSON** 
```json
{
    "title": "My dashboard",
    "version": "2.3",
    .......
    "pages": [
      {
        "title": "Main Page",
        "groups": [
          {
            "groupId": 1, // Group Id
            "border": "5px solid",
            "borderColor": "red"
          }
        ],
        "reports": [
          {
            "id": "322e3326-29f9-42d5-972b-e0b0b78ca581",
            "groupId": 1, // Group Id
            "groupOrder": 1, // Group order is used to achieve horizontal compaction
         .....
          },{
            "id": "483908d-29f9-42d5-972b-e0b0b78ca581",
            "groupId": 1, // Group Id
            "groupOrder": 2, // Group order is used to achieve horizontal compaction and it is incremental
         .....
          },
     ]
}
```

NOTICE
The program was tested solely for our own use cases, which might differ from yours.

Author Info: Monish [monish.nandakumaran@mercedes-benz.com](mailto:monish.nandakumaran@mercedes-benz.com) on behalf of Mercedes-Benz Research and Development India

[https://github.com/mercedes-benz/mercedes-benz-foss/blob/master/PROVIDER_INFORMATION.md](https://github.com/Daimler/daimler-foss/blob/master/PROVIDER_INFORMATION.md)

**Sample JSON**
[sample-moview-json.txt](https://github.com/neo4j-labs/neodash/files/12452338/sample-moview-json.txt)

Signed off: Monish <monish.nandakumaran@mercedes-benz.com>

alfredorubin96 commented:
Thank you for your work, could you please:

- [ ] add cypress test for this functionality
- [ ] attach a video of how it works?

m-o-n-i-s-h commented:
@alfredorubin96 , Added a GIF for your reference. This feature helps us to group two or more reports in a page.

```
// This is a sample dashboard json payload

{
  "title": "My dashboard",
  "version": "2.4",
  "settings": {
    "pagenumber": 0,
    "editable": true,
    "fullscreenEnabled": false,
    "parameters": {},
    "theme": "light"
  },
  "pages": [
    {
      "title": "Main Page",
      "groups": [
        {
          "groupId": 1,
          "border": "5px solid",
          "borderColor": "red"
        }
      ],
      "reports": [
        {
          "id": "50bbc9f4-dbec-4e41-8471-62b5a091b883",
          "title": "Not a group report",
          "query": "MATCH (n)-[e]->(m) RETURN n,e,m LIMIT 20\n\n\n",
          "width": 6,
          "height": 4,
          "x": 0,
          "y": 4,
          "type": "graph",
          "selection": {
            "Person": "name",
            "ACTED_IN": "(label)",
            "Movie": "title",
            "DIRECTED": "(label)",
            "PRODUCED": "(label)"
          },
          "settings": {},
          "schema": [
            [
              "Person",
              "born",
              "name"
            ],
            [
              "ACTED_IN",
              "roles"
            ],
            [
              "Movie",
              "tagline",
              "title",
              "released"
            ],
            [
              "DIRECTED"
            ],
            [
              "PRODUCED"
            ]
          ]
        },
        {
          "id": "6060fb1d-978b-429a-a020-3de2634cc711",
          "title": "Some random report",
          "query": "MATCH (n)-[e]->(m) RETURN n,e,m LIMIT 20\n\n\n",
          "width": 6,
          "height": 4,
          "x": 6,
          "y": 4,
          "type": "graph",
          "selection": {
            "Person": "name",
            "ACTED_IN": "(label)",
            "Movie": "title",
            "DIRECTED": "(label)",
            "PRODUCED": "(label)"
          },
          "settings": {
            "minimizable": true
          },
          "schema": [
            [
              "Person",
              "born",
              "name"
            ],
            [
              "ACTED_IN",
              "roles"
            ],
            [
              "Movie",
              "tagline",
              "title",
              "released"
            ],
            [
              "DIRECTED"
            ],
            [
              "PRODUCED"
            ]
          ]
        },
        {
          "id": "6a68555a-f891-493b-851a-dc15a34d948d",
          "groupId": 2,
          "groupOrder": 1,
          "title": "Group report 2 - A",
          "query": "MATCH (p:Person)-[:ACTED_IN]->(m:Movie)\nRETURN p, m.title",
          "width": 6,
          "height": 4,
          "x": 12,
          "y": 0,
          "type": "graph",
          "selection": {
            "Person": "name"
          },
          "settings": {
            "minimizable": true
          },
          "schema": [
            [
              "Person",
              "born",
              "name"
            ]
          ]
        },
        {
          "id": "8e632236-445b-483e-9540-627d3c8d34b7",
          "groupId": 2,
          "groupOrder": 2,
          "title": "Group report 2 - B",
          "query": "MATCH (p:Person)-[:ACTED_IN]->(m:Movie)\nRETURN p, collect(m.title)\n\n",
          "width": 6,
          "height": 4,
          "x": 18,
          "y": 0,
          "type": "graph",
          "selection": {
            "Person": "name"
          },
          "settings": {
            "minimizable": true
          },
          "schema": [
            [
              "Person",
              "born",
              "name"
            ]
          ]
        },
        {
          "id": "322e3326-29f9-42d5-972b-e0b0b78ca581",
          "groupId": 1,
          "groupOrder": 1,
          "title": "Group report 1 - A",
          "query": "**This is your first dashboard!** \n \nYou can click (⋮) to edit this report, or add a new report to get started. You can run any Cypher query directly from each report and render data in a variety of formats. \n \nTip: try _renaming_ this report by editing the title text. You can also edit the dashboard header at the top of the screen.\n\n\n",
          "width": 6,
          "height": 4,
          "x": 0,
          "y": 0,
          "type": "text",
          "selection": {},
          "settings": {
            "minimizable": true
          }
        },
        {
          "id": "c753770c-e1f7-41a7-b4d2-ddc418fcefb8",
          "groupId": 1,
          "groupOrder": 2,
          "title": "Group report 1 - B",
          "query": "MATCH (n)-[e]->(m) RETURN n,e,m LIMIT 20\n\n\n",
          "width": 6,
          "height": 4,
          "x": 6,
          "y": 0,
          "type": "graph",
          "selection": {
            "Person": "name",
            "ACTED_IN": "(label)",
            "Movie": "title",
            "DIRECTED": "(label)",
            "PRODUCED": "(label)"
          },
          "settings": {
            "minimizable": true
          },
          "schema": [
            [
              "Person",
              "born",
              "name"
            ],
            [
              "ACTED_IN",
              "roles"
            ],
            [
              "Movie",
              "tagline",
              "title",
              "released"
            ],
            [
              "DIRECTED"
            ],
            [
              "PRODUCED"
            ]
          ]
        }
      ],
      "toolbox": []
    }
  ],
  "parameters": {},
  "extensions": {
    "active": true,
    "activeReducers": []
  },
  "uuid": "bc58c8bf-d023-4528-8fc3-335d0d6d2d96"
}

```

![group report 1 example](https://github.com/neo4j-labs/neodash/assets/139316519/dea545df-2e7d-49cf-a2a5-568c1353de70)


![group report example 2](https://github.com/neo4j-labs/neodash/assets/139316519/88a86d6a-c165-46e5-8990-f517de66112e)

I have added a GIF for your reference. I have not implemented a settings for this. For now I am configuring it in JSON.

NOTICE
The program was tested solely for our own use cases, which might differ from yours.

Author Info: Monish [monish.nandakumaran@mercedes-benz.com](mailto:monish.nandakumaran@mercedes-benz.com) on behalf of Mercedes-Benz Research and Development India

[https://github.com/mercedes-benz/mercedes-benz-foss/blob/master/PROVIDER_INFORMATION.md](https://github.com/Daimler/daimler-foss/blob/master/PROVIDER_INFORMATION.md)

Sample JSON
[sample-moview-json.txt](https://github.com/neo4j-labs/neodash/files/12452338/sample-moview-json.txt)

Signed off: Monish [monish.nandakumaran@mercedes-benz.com](mailto:monish.nandakumaran@mercedes-benz.com)
